### ISSUE TITLE
Alternating background colours for tree items

### ISSUE THREAD

**Issue by AyyZerrAsa** [godotengine/godot-proposals#9376]
### Describe the project you are working on

Alternating colours to improve item visual separation in trees. A feature in Blender and Unreal's outliners that helps readability. 
| Before | After |
| ------------- | ------------- |
|![shortcuts_before](https://github.com/godotengine/godot-proposals/assets/138934452/80340f2a-be67-4959-994e-09fc64645fbc)|![shortcuts_after](https://github.com/godotengine/godot-proposals/assets/138934452/0eaaace3-827c-4f29-980f-d35d9201a1d3)|
|![tree_before](https://github.com/godotengine/godot-proposals/assets/138934452/d7f97b4d-ac9a-4def-98d5-3804a7550a47)|![tree_after](https://github.com/godotengine/godot-proposals/assets/138934452/3d80b559-e75a-4db6-a7bc-957334116d13)|

### Describe the problem or limitation you are having in your project

For long lists of items (especially the shortcut settings), it can be difficult to read due to there being no clear boundaries between items.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

A subtle alternating background colour on every other item clearly shows the boundary box of items.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Can be implemented by looping through the visible tree items. Can handle different item sizes if necessary. Empty space is filled by "standard" sized stripes.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

N/A

### Is there a reason why this should be core and not an add-on in the asset library?

This has to be implemented into core - needs to be drawn at a specific place in the tree code, and options need to be added to the theme cache.

**Comment by AyyZerrAsa** [comment]
Have the commit ready to make a pull request for if approved.

**Comment by KoBeWi** [comment]
If you have a commit ready, there is no reason to hold it. If you wanted to wait for approval, it's better to do so before completing your work.

**Comment by passivestar** [comment]
I really like the stripes in the shortcuts window, but I'm not sure I'm 100% sold on stripes in all of the docks. The thing is, Godot relies on trees a lot in its UI (in Blender for example it's just the Outliner), and having most of the UI covered with stripes could be distracting. Also notice there's currently a mismatch in the pattern when 2 docks are opened side by side:

<img width="1728" alt="image" src="https://github.com/godotengine/godot/assets/60579014/daa58c63-af3d-44db-9b4b-53f5ea68e09c"><br>

Maybe we need to consider docks separately from other things, there could be more value in stripes in the shortcuts window since it has many columns compared to docks that are narrow and don't have many columns

**Comment by KoBeWi** [comment]
I agree that it should not be enabled everywhere. It's better to have it opt-in (even to preserve some sort of compatibility, as making it enabled by default will affect all Trees people created). Here it even feels like it makes the list less readable:
![image](https://github.com/godotengine/godot-proposals/assets/2223172/8cef7ef1-b5d5-43b3-85f3-fcc8d2c85f93)

**Comment by AyyZerrAsa** [comment]
Agreed that the side-by-side looks gross 😄. In @passivestar's screenshot, the contents of the name/binding columns have a lot of whitespace between them, which is what makes the stripes more effective there.

**Comment by CarpenterBlue** [comment]
@passivestar 
> Also notice there's currently a mismatch in the pattern when 2 docks are opened side by side

I would say that's a feature because it helps determine Which dock you are looking at.
Truth to be told, I would prefer having the stripes everywhere rather than nowhere, it can be fixed later anyway.
